README.md

```diff
@@ -350,14 +367,22 @@ The only reserved ability MUST be the un-namespaced [`"*"` or "superuser"](#41-s
 
 ### 3.2.5 Proof of Delegation
 
-The `prf` field MUST contain UCAN witnesses (the "inputs" of a UCAN). As they need to be independently verifiable, proofs MUST either be the fully encoded version of a UCAN (including the signature) or the content address of the relevant proof. Attenuations not covered by a proof in the `prf` array MUST be treated as owned by the issuer DID.
+The `prf` field MUST contain the content address <!-- FIXME add link to CID discussion later in doc? --> of UCAN proofs (the "inputs" of a UCAN). Attenuations not covered by a proof in the `prf` array MUST be treated as <!-- FIXME as invalid, because we now have my: and as: --> owned by the issuer DID.
```

Comment:
Why would you self issue a UCAN? I'm not sure that I understand the use case. You already have access to all capabilities that you have. It's a noop.

Reply:
Fair enough let's say this ucan then

```ts
{
   iss: 'did:key:zAlice',
   aud: 'did:key:zService',
   att: { with: 'did:key:zAlice', can: '*' },
   prf: []
}
```

Maybe I'm misreading this, but I got an impression that would be invalid because we have `my:` and `as:`

Reply:
Yeah, in 0.8.x, you can express exactly that as

```js
{
   iss: 'did:key:zAlice',
   aud: 'did:key:zService',
   att: { with: 'my:*', can: '*' },
   prf: []
}
```

The resource you're working with is not the DID itself (e.g. add PK / revoke PK), but the capabilities is has by parenthood. Hence the distinction with `my:*`, which is "all of my capabilities", which also extends to `my:email`, `my:storage`, and so on.

Reply:
I mean, nothing is stopping you from defining a custom capability that works the way from your example. But this is the built-in equivalent designed to avoid that ambiguity.

Reply:
Ah my example still wasn't great one, this would be a better one, basically it's invocation to add specific CID to the given did (representing a set).

```ts
{
   iss: 'did:key:zAlice',
   aud: 'did:key:zService',
   att: { with: 'did:key:zAlice', can: 'store/add', link: 'bafy...hash' },
   prf: []
}
```

Seems like that is valid and I should not need `my` anywhere here.

Reply:
Just for my own understanding: what are you `add`ing? Is this "with my data store, you can add this link"?


Is that roughly equivalent to:

```
{
   iss: 'did:key:zAlice',
   aud: 'did:key:zService',
   att: { with: 'my:storage', can: 'store/add', link: 'bafy...hash' },
   prf: []
}
```

I think I need to re-read your "storage v2" doc, because I'm reading the above as being the RPC idea. (Which should probably get expanded into a dedicated spec).

Reply:
> Just for my own understanding: what are you adding to? The implicit storage resource for the user? Is this "with my data store, you can add this link"?

Idea is that data sets are identified via DIDs and users could add / remove links to the set. Users create/own those DIDs so they implicitly grant themself capabilities and in the example above invoke such capability to add new link to a set.

> Is that roughly equivalent to:

I don't think it is because user (or account rather account) may have multiple sets and they add links to specific one. Although accounts overal dataset is union of all sets, we do want to capture which specific set link is added to.

Reply:
> Idea is that data sets are identified via DIDs and users could add / remove links to the set

>  Although accounts overal dataset is union of all sets, we do want to capture which specific set link is added to.

Gotcha! In this case, the resource is that pinset, not the DID. Something like:

```json
{
   iss: 'did:key:zAlice',
   aud: 'did:key:zService',
   att: { with: 'storage://pinset/foo', can: 'store/add', link: 'bafy...hash' },
   prf: []
}
```